### PR TITLE
Rework Chrome MacOS installation to fix issues related to SHA

### PR DESCRIPTION
**Pull request by nikita-bykov**
# Description
Reworked Chrome installation by changing value of SHA256 in google-chrome cask file.
The `:no_check` value of SHA256 allows to ignore SHA check. It can fix possible issues during installation.

#### Related issue:
Rework Chrome MacOS installation to fix issues related to SHA [#1309](https://github.com/actions/virtual-environments-internal/issues/1309)
## Check list
- [x] Related issue / work item is attached
- [x] Changes are tested and related VM images are successfully generated
## Builds
 [macOS 10.13](https://github.visualstudio.com/virtual-environments/_build/results?buildId=86090&view=results), [macOS 10.14](https://github.visualstudio.com/virtual-environments/_build/results?buildId=86092&view=results), [macOS 10.15](https://github.visualstudio.com/virtual-environments/_build/results?buildId=86093&view=results), [macOS 11.0](https://github.visualstudio.com/virtual-environments/_build/results?buildId=86094&view=results)

### CODE REVIEW

File: images/macos/provision/utils/utils.sh [hunk 1]
```diff
@@ -96,4 +96,15 @@ get_default_xcode_from_toolset() {
 verlte() {
     sortedVersion=$(echo -e "$1\n$2" | sort -V | head -n1)
     [  "$1" = "$sortedVersion" ]
+}
+
+brew_install_ignoring_sha256() {
+    local TOOL_NAME=$1
+
+    CASK_DIR=/usr/local/Homebrew/Library/Taps/homebrew/homebrew-cask/Casks
```
Comment (maxim-lobanov) [inline images/macos/provision/utils/utils.sh:104]: Let's avoid hardcoding this path. Use `brew --repo homebrew/cask`

File: images/macos/provision/utils/utils.sh [hunk 2]
```diff
@@ -96,4 +96,15 @@ get_default_xcode_from_toolset() {
...
+    chmod a+w "$CASK_DIR/$TOOL_NAME.rb"
+    SHA=$(grep "sha256" "$CASK_DIR/$TOOL_NAME.rb" | awk '{print $2}')
+    sed -i '' "s/$SHA/:no_check/" "$CASK_DIR/$TOOL_NAME.rb"
+    brew cask install $TOOL_NAME
+    sed -i '' "s/:no_check/:$SHA/" "$CASK_DIR/$TOOL_NAME.rb"
```
Comment (maxim-lobanov) [inline images/macos/provision/utils/utils.sh:109]: Can we do `git reset` here?

File: images/macos/provision/utils/utils.sh [hunk 3]
```diff
@@ -96,4 +96,15 @@ get_default_xcode_from_toolset() {
 verlte() {
     sortedVersion=$(echo -e "$1\n$2" | sort -V | head -n1)
     [  "$1" = "$sortedVersion" ]
+}
+
+brew_install_ignoring_sha256() {
```
Comment (maxim-lobanov) [inline images/macos/provision/utils/utils.sh:101]: Can this function be unified to support both casks and formulas?

File: images/macos/provision/core/chrome.sh [hunk 4]
```diff
@@ -1,5 +1,7 @@
+source ~/utils/utils.sh
+
 echo "Installing Chrome..."
-brew cask install google-chrome
+brew_install_ignoring_sha256 "google-chrome"
```
Comment (pagejess009) [inline images/macos/provision/core/chrome.sh:4]:
```suggestion
brew_install_ignoring_sha256 "google-chrome"
```

**Review by pagejess009 — APPROVED** [review]

File: images/macos/provision/utils/utils.sh (hunk 1 above)
Comment (nikita-bykov) [reply images/macos/provision/utils/utils.sh]: Fixed it

File: images/macos/provision/utils/utils.sh (hunk 2 above)
Comment (nikita-bykov) [reply images/macos/provision/utils/utils.sh]: Fixed it

**Review by MaksimZhukov — APPROVED** [review]

**Review by maxim-lobanov — APPROVED** [review]

**Review by vsafonkin — APPROVED** [review]

**Review by al-cheb — APPROVED** [review]

**Pull request merged**